projects/storefrontlib/cms-components/myaccount/consent-management/components/consent-form/consent-management-form.component.ts

```diff
@@ -59,7 +87,9 @@ export class ConsentManagementFormComponent implements OnInit {
     this.consentGiven = !this.consentGiven;
 
     this.consentChanged.emit({
-      given: this.consentGiven,
+      given: this.useGetterForIsConsentGiven
```

Comment:
Is using a getter going to have performance impacts? As in, will the function be repeatedly called to check the consent? 
I think using a`BehaviorSubject` or `Observable` could be a better option.

Reply:
The issue here is that this property is a derivative state from the `@Input consent` and theoretically we can either set `this.consentGiven` inside `ngOnChanges` where we would react to the changes for the inputs(we can even push these values to the subject) or use this getter which will automatically return updated value based on current input. Since there are no heavy calculations, or any calculation at all for that matter I've decided to go with the getter approach. 
To reduce number of calls we can create a pipe which will only be fired if the input changes, and not on each re-render, but for implementing ngOnChanges and push values to the subject just doesn't sound like a very clean approach, that's just my opinion - or did you have something else in mind? Please let me know what do you think which approach is the best here

Reply:
Thank you for highlighting the reasoning and considerations here. I agree that the "getter" method is clean and responsive and I would really love for it to be that simple. But I think we need to emphasise performance as better practice. I think [this approach with signals](https://blog.angular-university.io/angular-signal-inputs/) would be the best solution to our situation. But we might need to fall back to the ngOnChanges method until next major when we support a later version of angular.

Reply:
I agree, I've applied `ngOnChanges` approach